packages/mobile/src/hooks/useLoadOfflineTracks.ts

```diff
@@ -0,0 +1,89 @@
+import type { Track, UserMetadata, UserTrackMetadata } from '@audius/common'
+import {
+  FeatureFlags,
+  Kind,
+  makeUid,
+  cacheActions,
+  savedPageTracksLineupActions
+} from '@audius/common'
+import moment from 'moment'
+import { useDispatch } from 'react-redux'
+import { useAsync } from 'react-use'
+
+import { useFeatureFlag } from 'app/hooks/useRemoteConfig'
+import { loadTracks } from 'app/store/offline-downloads/slice'
+
+import {
+  getTrackJson,
+  listTracks,
+  verifyTrack
+} from '../services/offline-downloader/offline-storage'
+
+export const useLoadOfflineTracks = async (collection: string) => {
```

Comment:
Sorry we def talked about this before, but remind me why we're putting this in a hook instead of a saga?

Reply:
We didn't really have a good reason one way or the other. Personally, I think it's easier to reason about the hooks since they get called very directly from the components that rely on them. However, we use sagas so much everywhere that I'm not sure it's worth introducing a new pattern now.